frontend/src/scenes/session-recordings/components/AiRegexHelper/AiRegexHelper.tsx

```diff
@@ -0,0 +1,63 @@
+/**
+ * @fileoverview A component that helps you to generate regex for your settings using Max AI
+ */
+
+import { IconCopy } from '@posthog/icons'
+import { LemonBanner, LemonButton, LemonModal, LemonTextArea } from '@posthog/lemon-ui'
+import { useActions, useValues } from 'kea'
+
+import { aiRegexHelperLogic } from './aiRegexHelperLogic'
+
+export function AiRegexHelper(): JSX.Element {
+    const filterLogic = aiRegexHelperLogic()
+    const { isOpen, input, generatedRegex, error, isLoading } = useValues(filterLogic)
+    const { setInput, handleGenerateRegex, handleApplyRegex, onClose } = useActions(filterLogic)
+
+    return (
+        <>
+            <LemonModal isOpen={isOpen} onClose={onClose} title="Max AI Regex Helper">
+                Explain your regex in natural language:
+                <LemonTextArea
+                    placeholder="I want an regex that covers all urls that include 'app.posthog.com/auth/*'"
+                    className="w-full my-2"
+                    maxRows={4}
+                    minRows={2}
+                    value={input}
+                    onChange={(value) => setInput(value)}
+                />
+                <div className="flex justify-center mt-2">
```

Comment:
should we have a "the only data we send to any third party is what you enter into the box", or similar?

hmmmm maybe we should have a proper setting area for all of this 🤔

Reply:
asked in the AI project manager channel in case they've already thought about this